src/CommunityToolkit.Maui/Layouts/StateContainer/StateContainer.shared.cs

```diff
@@ -27,12 +27,24 @@ public static class StateContainer
 	public static readonly BindableProperty ShouldAnimateOnStateChangeProperty
 		= BindableProperty.CreateAttached("ShouldAnimateOnStateChange", typeof(bool), typeof(StateContainer), true, propertyChanged: OnShouldAnimateOnStateChangeChanged);
 
+	/// <summary>
+	/// Backing BindableProperty for the <see cref="GetCanStateChange"/> method.
+	/// </summary>
+	public static readonly BindableProperty CanStateChangeProperty
```

Comment:
If I'm not wrong, the user can override this to another BindingMode, and then this will not be readonly anymore... In order to archive that, you need to use the `BindablePropertyKey` and expose the `BindableProperty` inside it. You can see an example [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/CameraView.shared.cs#L20-L31).
FYI, the sample code uses a normal BP so you need to convert that to be attached.

Reply:
Good catch! 

The difference in this use-case is that we still want to be able to change the value of `CanStateChangeProperty` internally; we just want to disallow devs using the `CommunityToolkit.Maui` NuGet Package to change the value of `CanStateChangeProperty` externally.

By using `BindingMode.OneWayToSource` when creating `CanStateChangeProperty`, we ensure that developers cannot change its value via ViewModel Bindings. I've included Unit Tests that ensure + verify this.

And by making `SetCanStateChange()` a private method, devs are unable to change `CanStateChange` via C# / code-behind; only we can call this method internally:

```cs
static void SetCanStateChange(BindableObject b, bool value) => b.SetValue(CanStateChangeProperty, value);
```

I did play around with `BindablePropertyKey` a bit, but found that we ultimately cannot use it. If I change this to a `BindablePropertyKey` using `BindableProperty.CreateReadOnly()`, then .NET MAUI will throw an `InvalidOperationException` whenever we call `private static SetCanStateChange()`